src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -128,4 +129,46 @@ public boolean supportsFolder() {
         return true;
     }
 
+
+    /**
+     * @deprecated Use <tt>GlobalConfigFiles.get().getConfigs()</tt> instead.
+     */
+    @Deprecated
+    public Collection<Config> getAllConfigs() {
```

Comment:
I like this approach. It would also restore compatibility for Job DSL.

But `getAllConfigs()` should only return the configs managed by this provider. See https://github.com/jenkinsci/config-file-provider-plugin/blob/config-file-provider-2.13/src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java#L77-L82. This change would return all global configs.